### PR TITLE
Add 'mixer build delta-packs' when using --new-swupd

### PR DESCRIPTION
**Pull request by cmarcelo**
This is on top of #79.

### CODE REVIEW

**Comment by cmarcelo** [comment]
The two last commits are the ones for this PR, others come from #79.

File: builder/builder.go [hunk 1]
```diff
@@ -939,3 +939,128 @@ func createCompressedArchive(dst string, srcs ...string) (err error) {
...
+
+	if to == 0 {
+		to = mixVersion
+	} else {
+		if to > mixVersion {
+			return errors.Errorf("--to version must be at most the latest mix version (%d)", mixVersion)
+		}
+	}
+	if from >= to {
+		return errors.Errorf("the --from version must not be larger than the --to version")
```
Comment (matthewrsj) [inline builder/builder.go]: "must not be larger than **or equal to** the --to version"

File: builder/builder.go [hunk 2]
```diff
@@ -650,20 +651,42 @@ func (b *Builder) buildUpdateWithNewSwupd(timer *stopWatch, mixVersion uint32, m
 	for _, f := range mom.Files {
 		fmt.Printf("- %-20s %d\n", f.Name, f.Version)
 	}
-	if !skipSigning {
+
+	if skipSigning {
+		fmt.Println("SKIPPING manifest signing.")
```
Comment (tmarcu) [inline builder/builder.go]: Don't need to print this, just remove this if IMHO.

File: builder/builder.go [hunk 3]
```diff
@@ -852,3 +875,66 @@ func parseUint32(s string) (uint32, error) {
 	}
 	return uint32(parsed), nil
 }
+
+func createCompressedArchive(dst string, srcs ...string) (err error) {
```
Comment (tmarcu) [inline builder/builder.go]: Should this be exported so we can use it elsewhere in the code?

File: builder/builder.go (hunk 1 above)
Comment (pohly) [reply builder/builder.go]: It isn't clear what gets negated by  "not": "not (larger than or equal)" or "(not larger) or equal" - isn't language wonderful? ;-}

I think this is simpler: "the --from version must be smaller than the --to version".

File: builder/builder.go (hunk 2 above)
Comment (cmarcelo) [reply builder/builder.go]: Removed this from the commit that added it (and other PR) and from this one.

File: builder/builder.go (hunk 3 above)
Comment (cmarcelo) [reply builder/builder.go]: This function wasn't written with the intent of being exported. If we see value later in exporting it, we can revisit it to make it so.

File: builder/builder.go (hunk 1 above)
Comment (cmarcelo) [reply builder/builder.go]: Yes, the "smaller" reads better. Fixed.

File: mixer/cmd/build.go [hunk 4]
```diff
@@ -137,6 +137,69 @@ var buildImageCmd = &cobra.Command{
...
+}
+
+func runBuildDeltaPacks(cmd *cobra.Command, args []string) error {
+	if !builder.UseNewSwupdServer {
+		// TODO: Depending on how long we are going to live with both implementations, it
+		// might be worth making this command call the script when not using new swupd-server.
+		failf("build delta packs is only available with --new-swupd\nUse mixer-pack-maker.sh instead.")
+	}
+
+	fromChanged := cmd.Flags().Changed("from")
```
Comment (tmarcu) [inline mixer/cmd/build.go]: Cool, haven't used Changed() before :)

**Pull request merged**